File: Leetcode/Typescript/2924. Find Champion II/solution.ts
// https://leetcode.com/problems/find-champion-ii/description

function findChampion(n: number, edges: number[][]): number {
    const inDegree: number[] = Array.from({ length: n }, () => 0);


    for (const [_, weak] of edges){
        inDegree[weak] += 1
    }

    let champCount = 0, champTeam = -1

    for (let team = 0; team < n; team++){
        if (inDegree[team] === 0){
            champTeam = team
            champCount += 1
        }
    }

    return champCount === 1 ? champTeam : -1
};

File: Leetcode/Python/2924. Find Champion II/solution.py
# https://leetcode.com/problems/find-champion-ii/description

class Solution:
    def findChampion(self, n: int, edges: List[List[int]]) -> int:
        in_degree = [0] * n

        for strong, weak in edges:
            in_degree[weak] += 1

        champ_count = 0
        champ_team = -1

        for team in range(n):
            if in_degree[team] == 0:
                champ_team = team
                champ_count += 1

        return champ_team if champ_count == 1 else -1
